filters/united-states/arizona/tempe/public-art-geojson-to-json.js

```diff
@@ -1,51 +1,86 @@
+
     if (typeof data === 'string' || data instanceof String) {
         data = JSON.parse(data);
     }
     
     let new_data = [];
+    let errors = [];
 
-    data.features?.map(data => {
-        let d = data.properties || {}
+    data.features?.map(d => {
         let item = {};
+        let skip = false;
         
-        item.name = d.Name;
-        if (item.name === undefined) {
-            console.log(`Data name not found for art with ID ${d.OBJECTID}`);
+        // add name (required)
+        if (!add_required(item, "name", d, d.properties.Name, errors)) {
+            skip = true;
         }
 
-        let coordinates = { longitude: d.Longitude, latitude: d.Latitude };
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with ID ${d.OBJECTID}`);
+        // add coordinates (required)
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, d.properties.Longitude, errors)) {
+            skip = true;
+        }
+        if (!add_required(coordinates, "latitude", d, d.properties.Latitude, errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
 
-        item.details = {
-          artist: d.Artist,
-          year: d.Year,
-          address: d.Address,
-          hours: d.Hours,
-          tab_name: d.Tab_Name,
-          short_desc: d.Short_Desc,
-          desc1: d.Desc1,
-          desc2: d.Desc2,
-          desc3: d.Desc3,
-          desc4: d.Desc4,
-          desc5: d.Desc5,
-          neighborhood: d.Neighborhood,
-          developer: d.Developer,
-          website: d.Website,
-          pic_url: d.Pic_URL,
-        }
+        //optional fields
+        // item.artist = d.properties.ARTIST_FIRST_NAME + " " + d.properties.ARTIST_LAST_NAME
+        add_if_not_null(item, "artist", d.properties.Artist);
+        add_if_not_null(item, "material", d.properties.Hours);
+        add_if_not_null(item, "area", d.properties.Neighborhood);
+        add_if_not_null(item, "image_url", [d.properties.Pic_URL]);
+
+        item.dates = create_dates_template();
+        add_if_not_null(item.dates.installed, "year", d.properties.Year_Sort_FirstYr);
+        remove_null_date_fields(item);
 
-        new_data.push(item);
+        item.address = {}
+        add_if_not_null(item.address, "street_address", d.properties.Address);
+        remove_if_empty(item, "address");
+
+        item.details = {}
+        add_if_not_null(item.details, "developer", d.properties.Developer);
+        add_if_not_null(item.details, "desc1", d.properties.Desc1);
+        add_if_not_null(item.details, "desc2", d.properties.Desc2);
+        add_if_not_null(item.details, "desc3", d.properties.Desc3);
+        add_if_not_null(item.details, "desc4", d.properties.Desc4);
+        add_if_not_null(item.details, "desc5", d.properties.Desc5);
+        remove_if_null(item.details);
```

Comment:
you may also want to include remove_if_empty(item, "details")

Reply:
These are always gonna be strings, so I think remove if null does the job.

Reply:
if no strings were added to details an empty details object could get through